docs/explanation/lifecycle.rst

```diff
@@ -2,7 +2,62 @@
 Lifecycle details
 *****************
 
+Each part is built in five separate steps:
```

Comment:
here's a suggestion from something that came up in a ROCKs discussion yesterday: users often confuse the directories that are involved in the build process, which they need to write a correct e.g. `build-override` script. Consider cross-referencing this section with the [Step Execution Environment](https://craft-parts.readthedocs.io/en/latest/reference/parts_steps.html#step-execution-environment) somehow; like saying `and the built artifacts are placed in $CRAFT_PART_INSTALL` on the `BUILD` description

Reply:
Considering that I was referencing that very section while writing this, that should've been obvious to me!

I've xref'd the parts steps and added a list of step outputs in the parts and steps reference.